src/common/schedulers/slurm_commands.py

```diff
@@ -46,134 +46,12 @@
-
-    def is_drained(self):
-        """
-        Check if slurm node is in drained state.
-
-        drained(sinfo) is equivalent to IDLE+DRAIN(scontrol) or DOWN+DRAIN(scontrol)
-        """
-        return self._is_drain() and (
-            self.SLURM_SCONTROL_IDLE_STATE in self.state or self.SLURM_SCONTROL_DOWN_STATE in self.state
-        )
-
-    def is_powering_down(self):
-        """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.state
-
-    def is_power(self):
-        """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.state
-
-    def is_down(self):
-        """Check if slurm node is in a down state."""
-        return self.SLURM_SCONTROL_DOWN_STATE in self.state and not self.is_powering_down()
-
-    def is_up(self):
-        """Check if slurm node is in a healthy state."""
-        return not self._is_drain() and not self.is_down() and not self.is_powering_down()
-
-    def is_powering_up(self):
-        """Check if slurm node is in powering up state."""
-        return self.SLURM_SCONTROL_POWER_UP_STATE in self.state
-
-    def is_online(self):
-        """Check if slurm node is online with backing instance."""
-        return self.state in self.SLURM_SCONTROL_ONLINE_STATES
-
-    def is_configuring_job(self):
-        """Check if slurm node is configuring with job and haven't begun to run a job."""
-        return self.is_powering_up() and self.has_job()
-
-    def is_power_with_job(self):
-        """Dynamic nodes allocated a job but power up process has not started yet."""
-        return self.state == self.SLURM_SCONTROL_POWER_WITH_JOB_STATE
-
-    def is_running_job(self):
-        """Check if slurm node is running a job but not in configuring job state."""
-        return not self.is_powering_up() and self.has_job() and not self.is_power_with_job()
-
-    def is_resume_failed(self):
-        """Check if node resume timeout expires."""
-        return self.state == self.SLURM_SCONTROL_RESUME_FAILED_STATE
-
-    def __eq__(self, other):
-        """Compare 2 SlurmNode objects."""
-        if isinstance(other, SlurmNode):
-            return self.__dict__ == other.__dict__
-        return False
-
-    def __repr__(self):
-        attrs = ", ".join(["{key}={value}".format(key=key, value=repr(value)) for key, value in self.__dict__.items()])
-        return "{class_name}({attrs})".format(class_name=self.__class__.__name__, attrs=attrs)
-
-    def __str__(self):
-        return f"{self.name}({self.nodeaddr})"
-
-
-class InvalidNodenameError(ValueError):
-    r"""
-    Exception raised when encountering a NodeName that is invalid/incorrectly formatted.
-
-    Valid NodeName format: {queue-name}-{st/dy}-{instancetype}-{number}
-    And match: ^([a-z0-9\-]+)-(st|dy)-([a-z0-9-]+)-\d+$
-    Sample NodeName: queue-1-st-c5xlarge-2
-    """
-
-    pass
-
-
 def parse_nodename(nodename):
```

Comment:
Add in _parse_node_info:
```
    node_info = slurm_node_info.split("---")
    slurm_nodes = []
    for node in node_info:
        lines = node.strip().splitlines()
        if lines:
            nodename_capture = re.match(r"^([a-z0-9\-]+)-(st|dy)-([a-z0-9]+)-\d+$", lines[0])
            # Valid NodeName format: {queue_name}-{st/dy}-{instancetype}-{number}
            if nodename_capture:
                queue_name, node_type, instance_name = nodename_capture.groups()
                if "st" == node_type:
                    node = StaticNode(*lines)
                    slurm_nodes.append(node)
                else:
                    node = DynamicNode(*lines)
                    slurm_nodes.append(node)
            else:
                log.warning("Ignoring node %s because it has an invalid name", lines[0])
    return slurm_nodes
```
Test case is also added

Reply:
why are we duplicating this logic both in _parse_node_info and parse_nodename? Can't _parse_node_info keep calling parse_nodename and properly handle the InvalidNodenameError exception?

Reply:
Yes, changed